BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -742,7 +756,7 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
             id<BOXURLSessionTaskDelegate> taskDelegate = [self taskDelegateForSessionId:session.configuration.identifier sessionTaskId:downloadTask.taskIdentifier];
             NSString *destinationFilePath = nil;
 
-            if ([taskDelegate conformsToProtocol:@protocol(BOXURLSessionDownloadTaskDelegate)]) {
+            if ([taskDelegate conformsToProtocol:@protocol(BOXURLSessionDownloadTaskDelegate)] && [taskDelegate respondsToSelector:@selector(destinationFilePath)]) {
```

Comment:
it could responds to selector if the selector happens to share the same name between 2 protocols, we want to make sure it has the right protocol first.